src/utils/searchAndSort.ts

```diff
@@ -0,0 +1,122 @@
+  const _value = value.replace(/[^0-9a-z]/gi, ' ').replace(/\s+/, ' ');
+
+  // Split into words, pad each word with spaces
+  const words = _value.split(' ').map((value) => ` ${value} `);
+
+  for (const word of words) {
+    position = 3;
+    // For some reason the postgres implementation also includes the first 2 characters of each word
+    trigrams.push(word.substring(0, 2));
+
+    // Split the words into trigrams
+    for (let i = 0; i < word.length; i += 1) {
+      if (position <= word.length) {
+        trigrams.push(word.substring(i, position).toLowerCase());
+        position += 1;
+      }
+    }
+  }
+
+  return trigrams;
+};
+
+const searchConditionMet = <T extends Record<string, unknown>>(result: T, condition: SearchNodePayload): boolean => {
+  // `as SearchNodePayload` casts below are because the SearchNodePayload in the generated types only has `operation`
+  // The the union type from our types has the correct properties
+  if (isNotNodePayload(condition)) {
+    return !searchConditionMet(result, condition.child as SearchNodePayload);
+  } else if (isAndNodePayload(condition)) {
+    return (condition.children as SearchNodePayload[]).every((_condition) => searchConditionMet(result, _condition));
+  } else if (isOrNodePayload(condition)) {
+    return (condition.children as SearchNodePayload[]).some((_condition) => searchConditionMet(result, _condition));
+  } else if (isFieldNodePayload(condition)) {
+    // Only 'Exact' and 'Fuzzy' condition types are supported
+    // `null` values (XYZ field contains no value) are also not supported
+    const resultValue = `${result[condition.field]}`.toLowerCase();
+    const searchValues = condition.values
+      .filter((value: string | null): value is string => value !== null)
+      .map((value) => value.toLowerCase());
+
+    if (condition.type === 'Exact') {
+      return searchValues.some((value) => resultValue.includes(value));
+    } else if (condition.type === 'Fuzzy') {
+      return searchValues.some((value) => {
+        // Split the search value into trigrams and see if the result field contains any trigram
+        const trigrams = splitTrigrams(value);
+        return trigrams.some((trigram: string) => resultValue.includes(trigram));
+      });
+    }
+  }
+
+  // This is not possible unless new node types are introduced
+  return false;
+};
+
+/**
+ * In-memory search (filter) and sort on a result list using the Search API search and sortOrder interfaces
+ * The search currently only supports `Exact` and 'Fuzzy' type searches, 'Range' and 'ExactOrFuzzy' are not supported
+ * If the result type contains number fields, those must be supplied in the `sortOrderConfig` if you wish to sort on them
+ * @param results         The list of results you want to search and sort
+ * @param search          (optional) The SearchNodePayload which contains filter conditions to apply to the results
+ * @param sortOrderConfig (optional) The sort order configuration contains:
```

Comment:
I tried to use the exact same structure as the Search API, but since it only deals with strings, and sometimes (like in the case of deliverables), the entity we want to sort will have numbers. This config allows the consumer to "tell" the function which fields are numbers so they can be sorted appropriately. 

I originally though about just checking to see if the field value was a number, but this would get complicated because the Search API search node payload accepts strings as values, even if they are numbers. So there might be a mismatch between the (hopefully untouched) search node payload structure and the data type on the entities that we are sorting. Another benefit of this approach is that we can additionally tell the function which fields _should_ be numbers, even if they are strings. This helps with sorting `SearchResponseElement`s which always deal in strings, even if the underlying field is supposed to be a number.

Reply:
This is cool. One suggestion is to extract out the sort functionality as a separate function and call it here. We can update the planting sites code to use the sort function.. just so we don't have similarish code scattered across. Does similar handling of number semantics on strings, etc.